uk.co.compucorp.civicrm.hrsampledata/CRM/HRSampleData/Upgrader.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/**
+ * Collection of upgrade steps.
+ */
+class CRM_HRSampleData_Upgrader extends CRM_HRSampleData_Upgrader_Base {
+
+  private $csvDir;
+
+  public function __construct($extensionName, $extensionDir)
+  {
+    parent::__construct($extensionName, $extensionDir);
+
+    $this->csvDir = $this->extensionDir . "/resources/csv";
+  }
+
+  public function install() {
+    $this->cleanTablesData();
+    $this->importSampleData();
+    $this->copyContactPhotos();
+  }
+
+  public function uninstall() {
+    $this->cleanTablesData();
+  }
+
+  /**
+   * Removes data from Specific tables
+   *
+   */
+  private function cleanTablesData() {
+    $this->executeSqlFile('sql/uninstall.sql');
+    $this->cleanCustomFieldsValues();
+    $this->cleanSampleDataValues();
+  }
+
+  /**
+   * Custom field values data need to be cleaned
+   * differently by getting all custom field tables
+   * from custom groups table and then cleaning
+   * each table one by one.
+   *
+   */
+  private function cleanCustomFieldsValues() {
+    $customFieldTables = civicrm_api3('CustomGroup', 'get', [
+      'sequential' => 1,
+      'table_name' => ['IS NOT NULL' => 1],
+      'return' => ["table_name"],
+      'options' => ['limit' => 0],
+    ])['values'];
+
+    foreach ($customFieldTables as $table) {
+      CRM_Core_DAO::executeQuery("TRUNCATE TABLE {$table['table_name']}");
+    }
+  }
+
+  /**
+   * Data such as option values, pay scales .. etc
+   * which are created by this extension are cleaned here.
+   *
+   */
+  private function cleanSampleDataValues() {
+    $filesToClean = [
+      'AbsenceType' => 'civicrm_hrabsence_type',
```

Comment:
yes we don't want to remove default values .. only the ones created by the sample data extension